src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -240,6 +240,18 @@ protected async Task<AuthenticationResult> CacheTokenResponseAndCreateAuthentica
         }
 
         private void ValidateAccountIdentifiers(ClientInfo fromServer)
```

Comment:
Not sure about this. I think the right fix is to bypass this check completely for anything else except `AcquireTokenSilent`.

Reply:
ok, I can update